Source/WebCore/inspector/InspectorInstrumentation.cpp

```diff
@@ -79,14 +80,26 @@
 #include <JavaScriptCore/InspectorDebuggerAgent.h>
 #include <JavaScriptCore/ScriptArguments.h>
 #include <JavaScriptCore/ScriptCallStack.h>
+#include <memory>
+#include <wtf/NeverDestroyed.h>
+#include <wtf/Observer.h>
 #include <wtf/StdLibExtras.h>
+#include <wtf/Vector.h>
+#include <wtf/WeakHashSet.h>
 
 namespace WebCore {
 
 using namespace Inspector;
 
 namespace {
 static HashSet<InstrumentingAgents*>* s_instrumentingAgentsSet = nullptr;
+#if ENABLE(WEBDRIVER_BIDI)
+static WeakHashSet<InspectorInstrumentationConsoleMessageObserver>& consoleMessageObservers()
+{
+    static NeverDestroyed<WeakHashSet<InspectorInstrumentationConsoleMessageObserver>> observers;
```

Comment:
do we actually want to support more than one observer?

Reply:
Especially after moving to the `AutomationInstrumentation` scheme, a single observer will be enough. I also renamed the methods from add/remove to set/clear.